folium/plugins/search.py

```diff
@@ -123,17 +122,9 @@ def __init__(
         self.placeholder = placeholder
         self.collapsed = collapsed
         self.options = remove_empty(**kwargs)
+        self.test_params()
```

Comment:
not sure if this is the same as testing these at render time. To be safe, I'd keep them at render time, since things may change between initialization and rendering.